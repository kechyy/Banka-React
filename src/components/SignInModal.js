import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { modalToggle, modalToggle2, closeModal } from '../store/modules/modals/actions';
import { authAction } from '../store/modules/auth/actions';
import checkUserType from '../utils/checkUserType'

class SignInModal extends Component {
  constructor() {
    super();
    this.handleInputFieldChange = this.handleInputFieldChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.isLoading = false;
    this.state = {
      submittting: false,
      email: '',
      password: '',
      errors: {
        email: '',
        password: ''
      }
    };
  }

  handleSubmit(e) {
    e.preventDefault();
    const { email, password } = this.state;
    const { authAction: loginAction } = this.props;
    const errors = {};
    if (!email) errors.email = 'Email cannot be empty';
    if (!password) errors.password = 'Password cannot be empty';
  
    if (Object.keys(errors).length) {
      this.setState(prevState => ({
        ...prevState,
        errors: {
          ...errors
        }
      }));
      return;
    }
    const {
      history
    } = this.props;
    const url = checkUserType();
    loginAction({ userData: { email, password, }, history, url });
  }

  handleInputFieldChange(event) {
    const {
      target: { name, value }
    } = event;
    this.setState(prevState => ({
      [name]: value,
      errors: {
        ...prevState.errors,
        [name]: ''
      }
    }));
  }
  render() {
    const {
      errors: { email: emailError, password: passwordError }
    } = this.state;
    const { signinDisplay, closeModal, modalToggle2 } = this.props
    return (
      <div className="modal" style={{display: signinDisplay}}>
        <div className="modal-content">
          <header>
            <a href="javascript:;" title="Close" className="modal-close" onClick={closeModal}>&times;</a>
            <h1>LOGIN HERE</h1>
            <p>Enter your login credentials to access this application</p>
          </header>
          <form onSubmit={this.handleSubmit}>
            <div className="modal-body">
              <div className="row loginForm">
                <label>Email Address</label>
                <div style={{width:'100%'}} >
                  <input type="email" className="input-block" name="email" required 
                  id="email" placeholder="Enter email address" onChange={this.handleInputFieldChange}/>
                  {!!emailError && (
                    <span className="text-red-600 text-xs">{emailError}</span>
                  )}
                </div>
                <label>Password</label>
                <div>
                  <input type="password" className="input-block" minLength="6" required name="password"
                   id="password" placeholder="Enter password" onChange={this.handleInputFieldChange} />
                </div>
                <div className="col-12-xs col-12-lg col-12-sm"><button className="btn btn-pink btn-full btnHover" type="submit">Sign in</button><br/><br/>
                  <a href="javascript:;" className="sign" onClick={modalToggle2}>Don't have an account? <span  className="btn-yellow-test">[SIGN UP]</span></a>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
        )
      }
  }
const mapStateToProps = state => ({
  signinDisplay: state.modalStatus.signinDisplay,
  page: state.modalStatus.page
});
const mapDispatchToProps = {
  modalToggle, modalToggle2, closeModal, authAction
}
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SignInModal));