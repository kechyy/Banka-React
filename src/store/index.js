import { createStore, applyMiddleware, compose } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from './modules';
import getUserInfo from '../utils/getUserInfo'

const loggerMiddleware = createLogger();
const storeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middlewares = [thunkMiddleware];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(loggerMiddleware);
}
const user = getUserInfo();
const store = createStore(
  rootReducer,
  {
    authReducer: {
      user,
      isAuthenticated: Boolean(user)
    }
  },
  storeEnhancers(applyMiddleware(...middlewares))
);

export default store;