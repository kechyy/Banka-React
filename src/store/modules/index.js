import { combineReducers } from 'redux';
import authReducer from './auth/reducers';
import modalStatus from './modals/reducer';
import profileReducer from './userProfile/reducer';
import accountReducer from './bankAccount/reducer';
import transactionReducer from './transaction/reducer'
import logoutReducer from './logout/reducer';


export default combineReducers({
  authReducer,
  modalStatus,
  profileReducer,
  accountReducer,
  logoutReducer,
  transactionReducer
});