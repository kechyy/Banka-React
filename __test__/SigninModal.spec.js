import React from 'react';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import store from '../src/store';
import { mount, shallow } from './enzyme';
import SignInModal from '../src/components/SignInModal';

const mockStore = configureMockStore([thunk]);
const renderWithEnzymes = state => {
  const store = mockStore({
    authReducer: {
      error: null,
      isAuthenticated: false,
      user: {},
      status: 'rest'
    },
    modalStatus: {
      signinDisplay: 'none',
      signupDisplay: 'none',
      page: '',
    },
    SignInModal: {
      ...state
    }
  });
  return mount(
    <Provider store={store}>
      <Router>
        <SignInModal {...props} />
      </Router>
    </Provider>
  );
};

const props = {
  authAction: jest.fn(),
  status: '',
  isAuthenticated: false,
  error: null,
  location: { url: '/user-dashboard' },
  history: { push: jest.fn() },
  closeModal:jest.fn(),
  modalToggle2:jest.fn(),
  modalToggle:jest.fn(),
  signinDisplay:'block',
  page:'signin'

};


describe('Signin component', () => {

  it('renders without crashing', () => {
    const wrapper = renderWithEnzymes();
    expect(wrapper).toBeDefined();
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper).toBeDefined();
    const field = wrapper.find('input').first();
    expect(field.exists()).toBe(true);
    expect(field.props().name).toEqual('email')

  });

  it('renders one form', () => {
    const wrapper = renderWithEnzymes();

    expect(wrapper.find('div')).toBeTruthy();
    expect(wrapper.find('h1').text()).toEqual('LOGIN HERE');
    expect(wrapper.find('form')).toBeTruthy();
    expect(wrapper.find('form').length).toBe(1);
  });

  it('renders 2 input fields and 1 submit button', () => {
    const wrapper = renderWithEnzymes();

    expect(wrapper.find('input').length).toBe(2);
    expect(wrapper.find('input').at(0).props().type).toBe('email');
    expect(wrapper.find('input').at(1).props().type).toBe('password');
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button').at(0).props().type).toBe('submit');

    
  });
});