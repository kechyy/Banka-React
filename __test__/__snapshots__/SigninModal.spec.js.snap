// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin component renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <withRouter(Connect(SignInModal))
        authAction={[MockFunction]}
        closeModal={[MockFunction]}
        error={null}
        history={
          Object {
            "push": [MockFunction],
          }
        }
        isAuthenticated={false}
        location={
          Object {
            "url": "/user-dashboard",
          }
        }
        modalToggle={[MockFunction]}
        modalToggle2={[MockFunction]}
        page="signin"
        signinDisplay="block"
        status=""
      >
        <Connect(SignInModal)
          authAction={[MockFunction]}
          closeModal={[MockFunction]}
          error={null}
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          isAuthenticated={false}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          modalToggle={[MockFunction]}
          modalToggle2={[MockFunction]}
          page="signin"
          signinDisplay="block"
          status=""
        >
          <SignInModal
            authAction={[Function]}
            closeModal={[Function]}
            error={null}
            history={
              Object {
                "action": "POP",
                "block": [Function],
                "createHref": [Function],
                "go": [Function],
                "goBack": [Function],
                "goForward": [Function],
                "length": 1,
                "listen": [Function],
                "location": Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                },
                "push": [Function],
                "replace": [Function],
              }
            }
            isAuthenticated={false}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            match={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            modalToggle={[Function]}
            modalToggle2={[Function]}
            page=""
            signinDisplay="none"
            status=""
          >
            <div
              className="modal"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <div
                className="modal-content"
              >
                <header>
                  <a
                    className="modal-close"
                    href="javascript:;"
                    onClick={[Function]}
                    title="Close"
                  >
                    Ã—
                  </a>
                  <h1>
                    LOGIN HERE
                  </h1>
                  <p>
                    Enter your login credentials to access this application
                  </p>
                </header>
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="modal-body"
                  >
                    <div
                      className="row loginForm"
                    >
                      <label>
                        Email Address
                      </label>
                      <div
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      >
                        <input
                          className="input-block"
                          id="email"
                          name="email"
                          onChange={[Function]}
                          placeholder="Enter email address"
                          required={true}
                          type="email"
                        />
                      </div>
                      <label>
                        Password
                      </label>
                      <div>
                        <input
                          className="input-block"
                          id="password"
                          minLength="6"
                          name="password"
                          onChange={[Function]}
                          placeholder="Enter password"
                          required={true}
                          type="password"
                        />
                      </div>
                      <div
                        className="col-12-xs col-12-lg col-12-sm"
                      >
                        <button
                          className="btn btn-pink btn-full btnHover"
                          type="submit"
                        >
                          Sign in
                        </button>
                        <br />
                        <br />
                        <a
                          className="sign"
                          href="javascript:;"
                          onClick={[Function]}
                        >
                          Don't have an account? 
                          <span
                            className="btn-yellow-test"
                          >
                            [SIGN UP]
                          </span>
                        </a>
                      </div>
                    </div>
                  </div>
                </form>
              </div>
            </div>
          </SignInModal>
        </Connect(SignInModal)>
      </withRouter(Connect(SignInModal))>
    </Router>
  </BrowserRouter>
</Provider>
`;
